homeassistant/components/sensor/dyson.py

```diff
@@ -43,12 +51,13 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 class DysonSensor(Entity):
     """Representation of Dyson sensor."""
 
-    def __init__(self, hass, device):
+    def __init__(self, hass, device, sensor_type):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.

Reply:
Ah, good to know. It's true that most of the sensors do not have `hass` in the `__init__()` signature (but a good fraction of them still has it...).

I'm happy to do the change, but just to note that both suggestions refer to the original code, not to this PR, and my attempt was to minimize the change.

Reply:
Since you're changing these lines, please clean it up.